MrTheGood/projectb_android
# Mock database

Changes:
 * Added a mock database
 * Added data classes

How to test: 
 1. Read code
 2. Try to figure out how a random memory is loaded every time the MemoriesFragment is shown. 

How to use database: 
```java
Database database = Database.getInstance();

// get memories list (for in the memories list page)
List<Memory> memories = database.getMemories();

// getting a single memory (requires already having an id)
Memory memory = database.findMemory(id);

// adding new memories
Memory memory = new Memory(id, location, date, title, description, null)
database.addMemory(memory);

// showing content (images not included in demo content)
dateTextView.setText(memory.getDateText());
titleTextView.setText(memory.getTitle());
descriptionTextView.setText(memory.getText());
imageView.setImageDrawable(memory.getThumbnail().getImage()); // for thumbnail in list
if (media instanceOf Image) imageView.setImageDrawable(media.getImage()); // for image in swipable detail list
```
